### PR TITLE
Hw jv jpa solution

### PR DESCRIPTION
**Pull request by odeineka**

### CODE REVIEW

File: src/main/java/com/example/demo/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -35,6 +35,27 @@ public BookDto findBookById(Long id) {
...
+        Book existingBook = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id " + id)
+        );
+
+        existingBook.setTitle(requestDto.getTitle());
+        existingBook.setAuthor(requestDto.getAuthor());
+        existingBook.setIsbn(requestDto.getIsbn());
+        existingBook.setPrice(requestDto.getPrice());
+        existingBook.setDescription(requestDto.getDescription());
+        existingBook.setCoverImage(requestDto.getCoverImage());
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/service/impl/BookServiceImpl.java:52]: Create method in mapper
```suggestion
        bookMapper.updateBookFromDto(requestDto, existingBook);
```

File: src/main/java/com/example/demo/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -35,6 +35,27 @@ public BookDto findBookById(Long id) {
...
+
+        return bookMapper.toDto(bookRepository.save(existingBook));
+    }
+
+    @Override
+    public void softDeleteBookById(Long id) {
+        bookRepository.deleteById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/service/impl/BookServiceImpl.java:59]: let's also check if book exists in db

File: src/main/java/com/example/demo/repository/BookRepository.java [hunk 3]
```diff
@@ -3,12 +3,15 @@
...
 
-    List<Book> getAll();
+    @Query("SELECT b FROM Book b WHERE b.isDeleted = false")
+    List<Book> findAllActive();
 
-    Optional<Book> findBookById(Long id);
-
-    Book createBook(Book book);
+    @Query("SELECT b FROM Book b WHERE b.id = :id AND b.isDeleted = false")
+    Optional<Book> findActiveById(@Param("id") Long id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepository.java:16]: remove unused method

File: src/main/java/com/example/demo/repository/BookRepository.java [hunk 4]
```diff
@@ -3,12 +3,15 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
-public interface BookRepository {
+public interface BookRepository extends JpaRepository<Book, Long> {
 
-    List<Book> getAll();
+    @Query("SELECT b FROM Book b WHERE b.isDeleted = false")
+    List<Book> findAllActive();
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepository.java:13]: remove unused method

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/example/demo/repository/BookRepository.java (hunk 3 above)
Comment (odeineka) [reply src/main/java/com/example/demo/repository/BookRepository.java:16]: The method were removed previously. If not sth went wrong with my PR

File: src/main/java/com/example/demo/repository/BookRepository.java (hunk 4 above)
Comment (odeineka) [reply src/main/java/com/example/demo/repository/BookRepository.java:13]: The method were removed previously.

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! But fix my comments before merging ;)

**Pull request merged**